contrib/babelfishpg_tsql/sql/sys_functions.sql

```diff
@@ -1222,7 +1222,11 @@ LANGUAGE plpgsql IMMUTABLE;
     but should keep using OPERATOR(sys.+) when input date is in datetimeoffset type.
 */
 CREATE OR REPLACE FUNCTION sys.dateadd_internal_df(IN datepart PG_CATALOG.TEXT, IN num INTEGER, IN startdate datetimeoffset) RETURNS datetimeoffset AS $$
+DECLARE
+	timezone INTEGER;
 BEGIN
+	timezone = sys.babelfish_get_datetimeoffset_tzoffset(startdate)::INTEGER * 2;
```

Comment:
this change adds timezone to start date but commit message says opposite, could you please explain

Reply:
Modified commit message.

Postgres outputs datetimeoffset data by removing/subtracting input timezone value from it.
Fix includes, working on the engine output to include timezone and thereby retain the input timestamp as is.
1> declare @dt datetimeoffset(6)
2> set @dt = '1912-10-25 12:24:32 +10:0'
3> select dateadd(month,1,@dt)
4> go
dateadd

 BABEL output: 1912-11-25 02:24:32.0000000 +10:00  (after emoving/subtracting input timezone value from it)

   SQL output: 1912-11-25 12:24:32.0000000 +10:00